addons/rating/models/rating.py

```diff
@@ -55,9 +70,20 @@ def _compute_rating_image(self):
     @api.multi
     @api.depends('rating')
     def _compute_rating_text(self):
-        text = {10: _('Satisfied'), 5: _('Not satisfied'), 1: _('Highly dissatisfied')}
+        text = {10: _('satisfied'), 5: _('not_satisfied'), 1: _('highly_dissatisfied')}
         for rating in self:
-            rating.rating_text = text[rating.rating] or _('No rating yet')
+            rating.rating_text = rating.rating and text[rating.rating] or _('no_rating')
+
+    @api.model
+    def create(self, vals):
+        if vals.get('res_model_id') and vals.get('res_id'):
+            model_name = self.env['ir.model'].browse(vals['res_model_id']).model
```

Comment:
Is it possible that this could trigger 'Access Error' ? Check if every user can read `ir.model`. Otherwise, a `sudo()` will be required.